### DISCUSSION TITLE
App Router inline CSS using `next/font`

### DISCUSSION THREAD

**Discussion by steve-marmalade** [vercel/next.js#71262] [App Router]
Using the Pages Router, it looks like NextJS will automatically inline CSS for fonts: https://nextjs.org/learn-pages-router/seo/improve/fonts

However, in the App Router, this does not appear to be happening; I am seeing a separate stylesheet that needs to be loaded, decreasing SEO performance metrics like FCP. I don't see any mentioning of CSS inlining in this guide: https://nextjs.org/learn/dashboard-app/optimizing-fonts-images

Does anyone know a workaround for how to inline CSS using next/font in the App Router? CC @samcx as you have been working on other CSS Inlining issues such as https://github.com/vercel/next.js/discussions/59989#discussioncomment-10194584 (FWIW I am using PPR + the latest Canary and I still see a > 500ms improvement to FCP if I manually inline the CSS, so Link Headers either isn't fully implemented or doesn't seem to be equivalently performant).

**Comment by leerob** [comment]
We're working on an experimental flag to enable CSS inlining, for both webpack/turbopack.

You can follow along on the PR: https://github.com/vercel/next.js/pull/72195
